pkg/cloud/services/ec2/launchtemplate.go

```diff
@@ -522,3 +524,28 @@ func (s *Service) getFilteredSecurityGroupID(securityGroup infrav1.AWSResourceRe
 
 	return *sgs.SecurityGroups[0].GroupId, nil
 }
+
+func getLaunchTemplateInstanceMarketOptionsRequest(spotMarketOptions *infrav1.SpotMarketOptions) *ec2.LaunchTemplateInstanceMarketOptionsRequest {
+	if spotMarketOptions == nil {
+		// Instance is not a Spot instance
+		return nil
+	}
+
+	// Set required values for Spot instances
+	spotOptions := &ec2.LaunchTemplateSpotMarketOptionsRequest{}
+
+	// Persistent option is not applicable EC2 autoscaling. If we do not specify a request type, EC2 makes a one-time request by default.
+	// Persistent requests are only supported when interruption behavior is set to either hibernate or stop.
+	// By default, interruption behavior is `terminate`, so no need to explicitly specify that here as well.
+
+	maxPrice := spotMarketOptions.MaxPrice
+	if maxPrice != nil && *maxPrice != "" {
+		spotOptions.SetMaxPrice(*maxPrice)
+	}
```

Comment:
```suggestion
	if maxPrice := aws.StringValue(spotMarketOptions.MaxPrice); maxPrice != "" {
		spotOptions.SetMaxPrice(maxPrice)
	}
```

Since you aren't using maxPrice we could use StringVaule like this. WDYT? :)

Reply:
Sounds good to me